test/Constraints/assignment.swift

```diff
@@ -40,7 +49,11 @@ i = j
 _ = (i, f)
 slice[7] = i
 
-slice[7] = f // expected-error{{cannot assign value of type 'Y' to type 'X'}}
+slice[7] = f // expected-error{{cannot assign value of type 'Y' to subscript of type 'X'}}
```

Comment:
@jrose-apple What do you think about updated message?

Reply:
I'm not sure if we use the phrasing "assign to a subscript" anywhere else, but I like that it's still pretty concise, so 👍. (In particular, "subscript with element type 'X'" would be more correct but I'm not sure it's worth it.)

Reply:
Thanks!